overlord/snapstate/storehelpers.go

```diff
@@ -693,17 +726,38 @@ func storeUpdatePlanCore(
 			return updatePlan{}, err
 		}
 
-		// TODO:COMPS: handle components here
+		// TODO:COMPS: for now, go back to the components that were already
+		// installed with this revision. to be more robust, we should refresh
+		// only the components that are installed with the current revision of
+		// the snap. this means we'll need to check with the store for which
+		// revisions now available for that snap.
+		compInfos, err := snapst.ComponentInfosForRevision(si.Revision)
+		if err != nil {
+			return updatePlan{}, err
+		}
+
+		components := make([]ComponentSetup, 0, len(compInfos))
+		for _, compInfo := range compInfos {
+			components = append(components, ComponentSetup{
+				CompSideInfo: &compInfo.ComponentSideInfo,
+				CompType:     compInfo.Type,
+			})
+		}
```

Comment:
it would feel a bit more balanced with the other case if this was a componentTargetsFromLocalRevision helper ?